include/cinatra/coro_http_client.hpp

```diff
@@ -222,15 +222,27 @@ class coro_http_client {
       }
       else {
         if (has_closed_) {
-          if (ec = co_await asio_util::async_connect(
-                  io_ctx_, socket_, u.get_host(), u.get_port());
-              ec) {
-            break;
+          if (!proxy_host_.empty() && !proxy_port_.empty()) {
+            if (ec = co_await asio_util::async_connect(
+                    io_ctx_, socket_, proxy_host_, proxy_port_);
+                ec) {
+              break;
+            }
+          }
+          else {
+            if (ec = co_await asio_util::async_connect(
+                    io_ctx_, socket_, u.get_host(), u.get_port());
+                ec) {
+              break;
+            }
```

Comment:
i think the if -else can be simplified, because handle_uri can deal with proxy, the return value is proxy host, port, try to improve it again.

Reply:
I try to improve it, but prepare_request_str must use raw host.
I can modify it to like follow:
```cpp
  std::string host = proxy_host_.empty() ? u.get_host() : proxy_host_;
  std::string port = proxy_port_.empty() ? u.get_port() : proxy_port_;
  if (ec = co_await asio_util::async_connect(
          io_ctx_, socket_,  u.get_host(), u.get_port());
      ec) {
      break;
  }
```

Reply:
Looks good